src/python/grpcio/grpc/_server.py

```diff
@@ -880,7 +880,11 @@ def _serve(state):
         if state.server_deallocated:
             _begin_shutdown_once(state)
         if event.completion_type != cygrpc.CompletionType.queue_timeout:
-            if not _process_event_and_continue(state, event):
+            try:
+                if not _process_event_and_continue(state, event):
+                    return
+            except RuntimeError:
+                _begin_shutdown_once(state)
```

Comment:
First time I use `_stop` to solve this problem, later I found out that `_begin_shutdown_once` can solve it too. 

Now, I used `_stop` instead of `_begin_shutdown_once`😺